src/mme-app/utils/mmeCommonUtils.cpp

```diff
@@ -269,17 +269,20 @@ SM::ControlBlock* MmeCommonUtils::findControlBlock(cmn::utils::MsgBuffer* buf)
 		}
 		case tau_request:
 		{
-			if(msgData_p->ue_idx > 0)
-				cb = SubsDataGroupManager::Instance()->findControlBlock(msgData_p->ue_idx);
-			
-			if (cb == NULL)
+			const struct tauReq_Q_msg &tau_Req = (msgData_p->msg_data.tauReq_Q_msg_m);
+			int cbIndex = SubsDataGroupManager::Instance()->findCBWithmTmsi(tau_Req.ue_m_tmsi);
+			if (cbIndex > 0)
 			{
-                            log_msg(LOG_INFO, "Failed to find control block using index %d."
-                                              " Allocate a temporary control block\n", msgData_p->ue_idx);
+				cb = SubsDataGroupManager::Instance()->findControlBlock(cbIndex);
+			}
+			else
+			{
+				log_msg(LOG_INFO, "Failed to find control block using mTmsi %d."
+                                              " Allocate a temporary control block\n", tau_Req.ue_m_tmsi);
 
-                            // Respond  with TAU Reject from default TAU event handler
-			    cb = SubsDataGroupManager::Instance()->allocateCB();
-			    cb->setTempDataBlock(DefaultMmeProcedureCtxt::Instance());
+                            	// Respond  with TAU Reject from default TAU event handler
+                            	cb = SubsDataGroupManager::Instance()->allocateCB();
+                            	cb->setTempDataBlock(DefaultMmeProcedureCtxt::Instance());
```

Comment:
Can you please elaborate how it works by creating temp block if we don't know anything about that tmsi ?

Reply:
When we are unable to find the control block for the given MTMSI, we allocate a temporary CB , through which we fire and add the event to the procedure queue, in the msgHandlers(s1MsgHandler in this case).
It is then redirected to the default action handler from where we respond with the corresponding reject message.